### PR TITLE
add takeover an existing cluster

### PR DESCRIPTION
**Pull request by fanux**

### CODE REVIEW

**Comment by shy-Xu** [comment]
no problem~I will implement this design

File: docs/design/takeover-already-running-cluster.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Takeover already existing cluster
+
+## Motivations
+
+If you already has a existing kubernetes cluster and you want using sealer to manage the cluster lifecycle, like join masters, scale up or down nodes, clean the cluster.
```
Comment (allencloud) [inline docs/design/takeover-already-running-cluster.md:5]: s/has/have
s/a/an
s/using/to use

File: docs/design/takeover-already-running-cluster.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Proposal
+
+1. Using `gen` command specify the nodes ssh password and kubeconfig file.
```
Comment (allencloud) [inline docs/design/takeover-already-running-cluster.md:9]: s/Using/Use
s/specify/to specify

File: docs/design/takeover-already-running-cluster.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+### Takeover a cluster
+
+```shell script
+sealer gen --passwd xxxx --kubeconfig ~/.kube/config --image kubernetes:v1.19.8
+```
+
+The takeover actually is generate a Clusterfile by kubeconfig.
+Sealer will call kubernetes API to get masters and nodes IP info, then generate a Clusterfile.
+
+Also sealer will pull a CloudImage witch match the kubernetes version.
```
Comment (allencloud) [inline docs/design/takeover-already-running-cluster.md:25]: s/witch match/which matches

File: docs/design/takeover-already-running-cluster.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Use cases
+
+### Takeover a cluster
+
+```shell script
+sealer gen --passwd xxxx --kubeconfig ~/.kube/config --image kubernetes:v1.19.8
+```
+
+The takeover actually is generate a Clusterfile by kubeconfig.
```
Comment (allencloud) [inline docs/design/takeover-already-running-cluster.md:22]: s/generate/to generate

**Review by shy-Xu — APPROVED** [review]

**Pull request merged**